backend/bom-bom-server/src/main/java/me/bombom/api/v1/reading/scheduler/ReadingScheduler.java

```diff
@@ -0,0 +1,26 @@
+package me.bombom.api.v1.reading.scheduler;
+
+import lombok.RequiredArgsConstructor;
+import me.bombom.api.v1.reading.service.ReadingService;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class ReadingScheduler {
+
+    private static final String TIME_ZONE = "Asia/Seoul";
+
+    private final ReadingService readingService;
+
+    @Scheduled(cron="0 0 0 * * *", zone = TIME_ZONE)
+    public void daily() {
+        readingService.updateContinueReadingCount();
+        readingService.resetTodayReadingCount();
+    }
+
+    @Scheduled(cron="0 0 0 * * MON", zone = TIME_ZONE)
+    public void weekly() {
+        readingService.resetWeeklyReadingCount();
+    }
+}
```

Comment:
p3

이 부분에서 3가지 의견이 있이서 이렇게 남겨요~

1. 메서드 명
지금은 괜찮지만 나중에 몇 달후 또는 다른 신입 멤버가 왔을 때 단순히 `readingScheduler.dayily()`만 보고 무슨 동작을 하는지 바로 파악하기 어려울 것 같아요 😭
해서 `resetDailyReadingCounts()`이런식으로 작성해서 메서드명만 보고 어떤 동작을 하는지 알게하면 어떨까요?

2. 상수화
cron 표현식은 한눈에 이해하기 어렵기 때문에 zone처럼 상수로  이름을 부여해주면 좋을 것 같아요 😊

3. 로그
운영 중에는 디버깅이 어렵고 특히 스케줄러는 특정시간에 동작하는거라 재현하기가 어려워 로깅을 남기면 더 좋을 것 같아요 😎


```suggestion
public class ReadingScheduler {

    private static final String TIME_ZONE = "Asia/Seoul";
    private static final String DAILY_CRON = "0 0 0 * * *";
    private static final String WEEKLY_CRON = "0 0 0 * * MON";

    private final ReadingService readingService;

    @Scheduled(cron = DAILY_CRON, zone = TIME_ZONE)
    public void resetDailyReadingCounts() {
        log.info("오늘 읽기 초기화 실행");
        readingService.resetTodayReadingCount();
    }

    @Scheduled(cron = WEEKLY_CRON, zone = TIME_ZONE)
    public void resetWeeklyReadingCount() {
        log.info("주간 읽기 초기화 실행");
        readingService.resetWeeklyReadingCount();
    }
}
```

Reply:
1.
메서드 이름을 `daily`로 했던 이유는 TodayReading count 뿐만 아니라 ContinueReading count 까지 리셋하기 때문에 이거를 TodayReading + ContinueReading 을 내포하는 이름을 짓기 모호해서였습니다. 이 두개를 내포하는 메서드 이름을 찾아서 짓거나, 만약 이 두가지에 대한 리셋 전략이 달라질 것 까지 고려한다면 아예 스케줄러 메서드를 나누는 것도 방법이 될 것 같아요.

2, 3. 
좋은 방법이네요 👍
로깅과 관련해서는, 특정 시간에 실제로 스케줄러가 실행되는지 테스트 코드로는 확인이 불가능하기 때문에, 로깅으로 모니터링하여 확인하는게 일반적이라고 해서 로깅이 필수적일 것 같아요.